mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1178,6 +1192,102 @@ pub fn send_maker_spends_taker_payment<T: UtxoCommonOps + SwapOps>(
     Box::new(fut.boxed().compat())
 }
 
+pub fn send_taker_spends_maker_payment_preimage<T: UtxoCommonOps + SwapOps>(
+    coin: T,
+    preimage: &[u8],
+    secret: &[u8],
+) -> TransactionFut {
+    let mut transaction: UtxoTx = try_tx_fus!(deserialize(preimage).map_err(|e| ERRL!("{:?}", e)));
+    let script = Script::from(transaction.inputs[0].script_sig.clone());
+
+    let instruction_1 = try_tx_fus!(script.get_instruction_at(0));
+    let instruction_2 = try_tx_fus!(script.get_instruction_at(instruction_1.step));
```

Comment:
Yes I thought the same. It also wasn't really necessary to check if the transaction has any outputs since the transaction is validated in the previous steps, but I added the checks anyways.